homeassistant/components/ssdp/__init__.py

```diff
@@ -100,19 +88,20 @@ class _SsdpServiceDescription:
     """SSDP info with optional keys."""
 
     ssdp_usn: str
-    ssdp_st: str
+    ssdp_st: str | None = None
```

Comment:
I have made this optional because in my opinion this is not full proof (maybe both `st` and `nt` are None):
```python
    # Increase compatibility.
    if not info.ssdp_st and info.ssdp_nt:
        info.ssdp_st = info.ssdp_nt
```

Reply:
Depending on the message type, either the ST (Search Target, from M-SEARCH messages) or NT (Notification Type, from NOTIFY messages) header is mandatory. So you'll always have one or the other, but probably not both. Their contents are pretty much identical too.

Reply:
I have adjusted the code accordingly.